File: advent-of-code/scala/2015_01.scala


val answer1: Int = input.map(c => if (c == '(') 1 else -1).sum
println(answer1)

@scala.annotation.tailrec
def solve(
    s: String,
    floor: Int,
    position: Int,
  ): Int =
  if (floor == -1) position
  else solve(s.tail, floor + (if (s.head == '(') 1 else -1), position + 1)

val answer2 = solve(input, 0, 0)
println(answer2)

File: advent-of-code/python/2015/2015_01.py
from aocd import get_data, submit


inp = get_data(day=1, year=2015)

answer1 = sum(1 if x == '(' else -1 for x in inp)
submit(str(answer1), part="a", day=1, year=2015)


# Part 2


floor = 0
for position, x in enumerate(inp, 1):
  floor += 1 if x == '(' else -1
  if (floor == -1):
    break
answer2 = position
submit(str(answer2), part="b", day=1, year=2015)
